src/redis.js

```diff
@@ -0,0 +1,39 @@
+const logger = require("./logger");
+const Redis = require("ioredis");
+const MockRedis = require("ioredis-mock");
+
+// If you need to set the Redis URL, do it in REDIS_URL
+const redisUrl = "http://redis:6379";
+
+// Set MOCK_REDIS=1 to mock, MOCK_REDIS= to use real redis
+const useMockRedis = process.env.MOCK_REDIS;
+
+// RedisConstructor is one of Redis or MockRedis
+const redisConstructor = useMockRedis ? MockRedis : Redis;
+
+function createRedisClient() {
+  try {
+    const { port, host } = new URL(redisUrl);
+    return new redisConstructor(port, host, {
+      password: process.env.REDIS_PASSWORD,
+    });
+  } catch (error) {
+    const message = `Unable to parse port and host from "${redisUrl}"`;
+    logger.error({ error }, message);
+    throw new Error(message);
+    //createError(error, message); Future Proofing when createError is merged.
```

Comment:
I left this commented out, until createError is merged. Once that happens, I'll be using that instead of throwing an Error.